eccrypto.h

```diff
@@ -327,8 +327,8 @@ CRYPTOPP_DLL_TEMPLATE_CLASS DL_PrivateKeyImpl<DL_GroupParameters_EC<ECP> >;
 CRYPTOPP_DLL_TEMPLATE_CLASS DL_PrivateKeyImpl<DL_GroupParameters_EC<EC2N> >;
 CRYPTOPP_DLL_TEMPLATE_CLASS DL_PrivateKey_EC<ECP>;
 CRYPTOPP_DLL_TEMPLATE_CLASS DL_PrivateKey_EC<EC2N>;
-CRYPTOPP_DLL_TEMPLATE_CLASS DL_Algorithm_GDSA<ECP::Point>;
-CRYPTOPP_DLL_TEMPLATE_CLASS DL_Algorithm_GDSA<EC2N::Point>;
+CRYPTOPP_DLL_TEMPLATE_CLASS DL_Algorithm_GDSA<ECP::Point, SHA256, false>;

```

Comment:
> What's the purpose of this code?

The short of it: its used to manually instantiate a template for the shared object or DLL.

Also see the discussion of `CRYPTOPP_DLL_TEMPLATE_CLASS` at [config.h](https://cryptopp.com/wiki/Config.h) on the Crypto++ wiki.

Also see [Explicit Instantiation](http://msdn.microsoft.com/en-us/library/by56e477%28v=vs.100%29.aspx) on MSDN and [ 7.5 Where's the Template?](http://gcc.gnu.org/onlinedocs/gcc/Template-Instantiation.html) in the GCC manual.